### PR TITLE
feat: add path blacklist configuration

### PR DESCRIPTION
**Pull request by hanxiii-z**
add path blacklist configuration in the setting interface

Log: users can add paths to the exclusion list to prevent searching in them

Task: https://pms.uniontech.com/task-view-110869.html
Change-Id: I5014d879557cc063b4822adac5447c2df832db34

### CODE REVIEW

File: src/grand-search/gui/searchconfig/blacklistview/blacklistdelegate.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef BLACKLISTDELEGATE_H
+#define BLACKLISTDELEGATE_H
+
+#include <DListView>
+
```
Comment (wangchunlin5013) [inline src/grand-search/gui/searchconfig/blacklistview/blacklistdelegate.h:25]: Invalid blank line

File: src/grand-search/gui/searchconfig/blacklistview/blacklistmodel.h [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+class BlackListModel : public QStandardItemModel
+{
+    Q_OBJECT
+public:
+    explicit BlackListModel(int rows, int columns, QObject *parent = Q_NULLPTR);
+    ~BlackListModel() override;
+
+    virtual Qt::ItemFlags flags(const QModelIndex &index) const override;
+    QStringList mimeTypes() const override;
+
```
Comment (wangchunlin5013) [inline src/grand-search/gui/searchconfig/blacklistview/blacklistmodel.h:37]: Invalid blank line

File: src/grand-search/gui/searchconfig/blacklistwidget.cpp [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    }
+}
+
+void BlackListWidget::selectedChanged(const QItemSelection &selected, const QItemSelection &deselected) const
+{
+    Q_UNUSED(selected)
+    Q_UNUSED(deselected)
+    m_deleteButton->setEnabled(!m_listWrapper->selectionModel()->selectedRows().isEmpty());
+}
+
```
Comment (wangchunlin5013) [inline src/grand-search/gui/searchconfig/blacklistwidget.cpp:126]: Invalid blank line

**Review by Kakueeen — APPROVED** [review]

**Review by wangchunlin5013 — DISMISSED** [review]

File: src/grand-search/gui/searchconfig/blacklistview/blacklistview.h [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+ */
+
+#ifndef BLACKLISTVIEW_H
+#define BLACKLISTVIEW_H
+
+#include <DListView>
+#include <DWidget>
+
+#include <QDropEvent>
+#include<QVBoxLayout>
```
Comment (itsXuSt) [inline src/grand-search/gui/searchconfig/blacklistview/blacklistview.h:29]: space

File: src/grand-search/gui/searchconfig/blacklistview/deletedialog.h [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef DELETEDIALOG_H
+#define DELETEDIALOG_H
+
+#include<DDialog>
```
Comment (itsXuSt) [inline src/grand-search/gui/searchconfig/blacklistview/deletedialog.h:25]: space

**Review by itsXuSt — CHANGES_REQUESTED** [review]

**Review by Kakueeen — APPROVED** [review]

**Review by Clauszy — APPROVED** [review]

**Pull request merged**